src/routes/balances/balances.service.ts

```diff
@@ -106,12 +126,130 @@ export class BalancesService {
     };
   }
 
+  private async _getFromSimpleBalances(args: {
+    chainId: string;
+    safeAddress: string;
+    fiatCode: string;
+    trusted?: boolean;
+    excludeSpam?: boolean;
+  }): Promise<Balances> {
+    const { chainId, fiatCode } = args;
+    const simpleBalances =
+      await this.balancesRepository.getSimpleBalances(args);
+    const { nativeCurrency } = await this.chainsRepository.getChain(chainId);
+    const balances: Balance[] = await Promise.all(
+      simpleBalances.map(async (balance) =>
+        this._mapSimpleBalance(balance, chainId, fiatCode, nativeCurrency),
+      ),
+    );
+    const fiatTotal = balances
+      .filter((b) => b.fiatBalance !== null)
+      .reduce((acc, b) => acc + Number(b.fiatBalance), 0);
+
+    return <Balances>{
+      fiatTotal: this.getNumberString(fiatTotal),
+      items: orderBy(balances, 'fiatBalance', 'desc'),
+    };
+  }
+
+  private async _mapSimpleBalance(
```

Comment:
This function contains code that is mostly a duplicate of the previously existing`mapBalance`, except for minor details. I opted to include an almost-duplicated function instead of merging it with the original one, so after the feature flag cleaning, it can be just deleted/renamed.